python/cudf/tests/test_dataframe.py

```diff
@@ -2169,6 +2169,69 @@ def test_reset_index(pdf, gdf, drop):
               gdf.x.reset_index(drop=drop))
 
 
+@pytest.mark.parametrize('copy', [True, False])
+def test_reindex(copy):
+    index = [-3, -2, 0, 1, 3, 4, 6]
+    columns = ['a', 'b', 'c', 'd', 'e']
+    # TODO (ptaylor): pandas changes `int` dtype to `float64`
+    # when reindexing and filling new label indices with NaN
+    gdf = gd.datasets.randomdata(nrows=6, dtypes={'a': 'category',
+                                                  # 'b': int,
+                                                  'c': float,
+                                                  'd': str})
+    pdf = gdf.to_pandas()
+
+    # Validate labels are used as index when axis defaults to 0
+    assert_eq(pdf.reindex(index, copy=True),
+              gdf.reindex(index, copy=copy))
+
+    # Validate labels are used as index when axis=0
+    assert_eq(pdf.reindex(index, axis=0, copy=True),
+              gdf.reindex(index, axis=0, copy=copy))
+
+    # Validate labels are used as columns when axis=0
+    assert_eq(pdf.reindex(columns, axis=1, copy=True),
+              gdf.reindex(columns, axis=1, copy=copy))
+
+    # Validate labels are used as index when axis=0
+    assert_eq(pdf.reindex(labels=index, axis=0, copy=True),
+              gdf.reindex(labels=index, axis=0, copy=copy))
+
+    # Validate labels are used as columns when axis=1
+    assert_eq(pdf.reindex(labels=columns, axis=1, copy=True),
+              gdf.reindex(labels=columns, axis=1, copy=copy))
+
+    # Validate labels are used as index when axis='index'
+    assert_eq(pdf.reindex(labels=index, axis='index', copy=True),
+              gdf.reindex(labels=index, axis='index', copy=copy))
+
+    # Validate labels are used as columns when axis='columns'
+    assert_eq(pdf.reindex(labels=columns, axis='columns', copy=True),
+              gdf.reindex(labels=columns, axis='columns', copy=copy))
+
+    # Validate reindexes labels when index=labels
+    assert_eq(pdf.reindex(index=index, copy=True),
+              gdf.reindex(index=index, copy=copy))
+
+    # Validate reindexes column names when columns=labels
+    assert_eq(pdf.reindex(columns=columns, copy=True),
+              gdf.reindex(columns=columns, copy=copy))
+
+    # Validate reindexes both labels and column names when
+    # index=index_labels and columns=column_labels
+    assert_eq(pdf.reindex(index=index, columns=columns, copy=True),
+              gdf.reindex(index=index, columns=columns, copy=copy))
```

Comment:
Instead of having all of these checked in a single test it would be better to parameterize the test with different kwargs so that if something fails it can be much more easily triaged.